packages/wonder-blocks-dropdown/src/components/dropdown-popper.js

```diff
@@ -0,0 +1,105 @@
+// @flow
+import * as React from "react";
+import ReactDOM from "react-dom";
+import {Popper} from "react-popper";
+
+import {maybeGetPortalMountedModalHostElement} from "@khanacademy/wonder-blocks-modal";
+
+import type {StyleType} from "@khanacademy/wonder-blocks-core";
+
+const modifiers = [
+    {
+        name: "preventOverflow",
+        options: {
+            rootBoundary: "viewport",
+            // Allows to overlap the popper in case there's no more vertical
+            // room in the viewport.
+            altAxis: true,
+            // Also needed to make sure the Popper will be displayed correctly
+            // in different contexts (e.g inside a Modal)
+            tether: false,
+        },
+    },
+];
+
+type Props = {|
+    /**
+     * The children that will be wrapped by PopperJS.
+     */
+    children: (isReferenceHidden: boolean) => React.Node,
+
+    /**
+     * The reference element used to position the popper.
+     */
+    referenceElement: ?HTMLElement,
+
+    /**
+     * Whether this menu should be left-aligned or right-aligned with the
+     * reference component. Defaults to left-aligned.
+     */
+    alignment?: "left" | "right",
+
+    /**
+     * The popper's reference.
+     * @see https://popper.js.org/react-popper/v2/render-props/#innerref
+     */
+    onPopperElement?: (popperElement: ?HTMLElement) => mixed,
+
+    /**
+     * Styles that will be applied to the children.
+     */
+    style?: StyleType,
+|};
+
+/**
+ * A wrapper for PopperJS that renders the children inside a portal.
+ */
+export default function DropdownPopper({
```

Comment:
Good call splitting this out into its own component.  dropdown-core.js is getting pretty complicated.

Reply:
Yep, and I even have more plans to reorganize this file (to be working on that on a separate ticket).